### PR TITLE
Add MvvmCross to iOS project

### PR DESCRIPTION
**Pull request by willsb**
This adds the necessary bootstrap files for a MvvmCross project in iOS. If you have any other doubts on the reasonings, feel free to ask me, but the core changes are:

- The `AppDelegate` now starts the MvvmCross navigation system.
- The `DebugTrace` is injected so we can use it in the ViewModels for logging. We should consider using [Serilog](https://serilog.net/) for this.
- We added Fody because we need a reference to the actual assembly for Bait-and-switch to work.
- `LinkerPleaseInclude` is used to prevent some methods from being stripped by the Linker.
- `Setup` is used for registering platform specific components.

### CODE REVIEW

**Comment by amulware** [comment]
If you'd like to make a case for Serilog, please create an issue for logging, and we can talk about it there.

File: Toggl.Daneel/LinkerPleaseInclude.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+using UIKit;
+
+namespace Toggl.Daneel
+{
+    // This class is never actually executed, but when Xamarin linking is enabled it does ensure types and properties
+    // are preserved in the deployed app
+    [Preserve(AllMembers = true)]
+    public class LinkerPleaseInclude
+    {
+		public void Include(MvxTaskBasedBindingContext c)
```
Comment (amulware) [inline Toggl.Daneel/LinkerPleaseInclude.cs]: Formatting

File: Toggl.Daneel/LinkerPleaseInclude.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        {
+            c.Dispose();
+            var c2 = new MvxTaskBasedBindingContext();
+            c2.Dispose();
+        }
+
+        public void Include(UIButton uiButton)
+        {
+            uiButton.TouchUpInside += (s, e) =>
+                                      uiButton.SetTitle(uiButton.Title(UIControlState.Normal), UIControlState.Normal);
```
Comment (amulware) [inline Toggl.Daneel/LinkerPleaseInclude.cs]: Please explain what these things are doing.

File: Toggl.Daneel/Services/ApiFactory.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace Toggl.Daneel.Services
+{
+    public class ApiFactory : IApiFactory
+    {
+        #if DEBUG
+        private const ApiEnvironment environment = ApiEnvironment.Staging;
+        #else
+        private const ApiEnvironment environment = ApiEnvironment.Production;
+        #endif
```
Comment (amulware) [inline Toggl.Daneel/Services/ApiFactory.cs]: While fine for now, I don't think we want the DEBUG/RELEASE distinction to control the api environment in the future. This needs to be discussed.

File: Toggl.Daneel/Setup.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace Toggl.Daneel
+{
+    public class Setup : MvxIosSetup
+    {
+#if DEBUG
+        private const ApiEnvironment environment = ApiEnvironment.Staging;
+#else
+        private const ApiEnvironment environment = ApiEnvironment.Production;
+#endif
```
Comment (amulware) [inline Toggl.Daneel/Setup.cs]: Different formatting of the # thingies. Also, selecting the api environment in multiple places? That smells fishy to me.

File: Toggl.Daneel/Info.plist [hunk 5]
```diff
@@ -1,38 +1,38 @@
...
-    </array>
-    <key>UISupportedInterfaceOrientations</key>
-    <array>
-        <string>UIInterfaceOrientationPortrait</string>
-        <string>UIInterfaceOrientationLandscapeLeft</string>
-        <string>UIInterfaceOrientationLandscapeRight</string>
-    </array>
-    <key>XSAppIconAssets</key>
-    <string>Assets.xcassets/AppIcon.appiconset</string>
+	<key>CFBundleName</key>
```
Comment (willsb) [inline Toggl.Daneel/Info.plist]: This was done by Visual Studio, so changing it will be of no use

File: Toggl.Daneel/LinkerPleaseInclude.cs (hunk 1 above)
Comment (willsb) [inline Toggl.Daneel/LinkerPleaseInclude.cs]: ✅

File: Toggl.Daneel/LinkerPleaseInclude.cs (hunk 2 above)
Comment (willsb) [inline Toggl.Daneel/LinkerPleaseInclude.cs]: This is being explained both in the file and in the PR description

Comment (amulware) [inline Toggl.Daneel/LinkerPleaseInclude.cs]: I don't see an explanation of why this subscribes, and why strings are replaced with themselves, etc.

**Pull request merged**